labscript_utils/splash.py

```diff
@@ -98,27 +94,15 @@ def show(self):
     def update_text(self, text):
         self.text = text
         self.label.setText(text)
-        # If we are not visible yet, exec until we are painted.
-        if not self._first_paint_complete:
-            self.qapplication.exec_()
-        else:
-            self.repaint()
+        self._paint_pending = True
+        while self._paint_pending:
+            QtCore.QCoreApplication.processEvents(QtCore.QEventLoop.AllEvents)
+            QtCore.QCoreApplication.sendPostedEvents()
```

Comment:
Did some debugging. The loop gets stuck here because the paint event never gets triggered from some reason. Adding `self.update()` just before entering the loop prevents the block for me, though I'm not sure it is the "correct" solution.